src/recipe/error.rs

```diff
@@ -0,0 +1,354 @@
+            kind: $kind,
+        }
+    }};
+}
+
+/// Macro to facilitate the creation of [`Error`]s.
+#[macro_export]
+#[doc(hidden)]
+macro_rules! _partialerror {
+    ($span:expr, $kind:expr $(,)?) => {{
+        $crate::recipe::error::PartialParsingError {
+            span: $span,
+            label: None,
+            help: None,
+            kind: $kind,
+        }
+    }};
+    ($span:expr, $kind:expr, label = $label:expr $(,)?) => {{
+        $crate::recipe::error::PartialParsingError {
+            span: $span,
+            label: Some($label),
+            help: None,
+            kind: $kind,
+        }
+    }};
+    ($span:expr, $kind:expr, help = $help:expr $(,)?) => {{
+        $crate::recipe::error::PartialParsingError {
+            span: $span,
+            label: None,
+            help: Some($help),
+            kind: $kind,
+        }
+    }};
+    ($span:expr, $kind:expr, label = $label:expr, help = $help:expr $(,)?) => {{
+        $crate::_error!($src, $span, $kind, $label, $help)
+    }};
+    ($span:expr, $kind:expr, help = $help:expr, label = $label:expr $(,)?) => {{
+        $crate::_error!($src, $span, $kind, $label, $help)
+    }};
+    ($span:expr, $kind:expr, $label:expr, $help:expr $(,)?) => {{
+        $crate::recipe::error::PartialParsingError {
+            span: $span,
+            label: Some($label),
+            help: Some($help),
+            kind: $kind,
+        }
+    }};
+}
+
+/// Error handler for [`marked_yaml::LoadError`].
+pub(crate) fn load_error_handler(src: &str, err: marked_yaml::LoadError) -> ParsingError {
+    _error!(
+        src,
+        marker2span(src, marker(&err)),
+        ErrorKind::YamlParsing(err),
+        label = match err {
+            marked_yaml::LoadError::TopLevelMustBeMapping(_) => "expected a mapping here",
+            marked_yaml::LoadError::UnexpectedAnchor(_) => "unexpected anchor here",
+            marked_yaml::LoadError::UnexpectedTag(_) => "unexpected tag here",
+            _ => "here",
+        }
+    )
+}
+
+/// Convert a [`marked_yaml::Marker`] to a [`SourceSpan`].
+#[inline(always)]
+pub(crate) fn marker2span(src: &str, mark: marked_yaml::Marker) -> SourceSpan {
+    let start = SourceOffset::from_location(src, mark.line(), mark.column());
+
+    SourceSpan::new(
+        start,
+        SourceOffset::from(find_length(src, start)), //::from_location(src, mark.line(), mark.column()),
+    )
+}
+
+/// Get the [`marked_yaml::Marker`] from a [`marked_yaml::LoadError`].
+#[inline(always)]
+pub(crate) fn marker(err: &marked_yaml::LoadError) -> marked_yaml::Marker {
+    use marked_yaml::LoadError::*;
+    match err {
+        TopLevelMustBeMapping(m) => *m,
+        UnexpectedAnchor(m) => *m,
+        MappingKeyMustBeScalar(m) => *m,
+        UnexpectedTag(m) => *m,
+        ScanError(m, _) => *m,
+    }
+}
+
+/// Convert a [`marked_yaml::Span`] to a [`SourceSpan`].
+pub(crate) fn markerspan2span(src: &str, span: marked_yaml::Span) -> SourceSpan {
```

Comment:
It doesn't. I did this on purpose at first, since it is for internal usage only, and it should remain for internal `recipe` usage only. I usually do this because it makes it clear to the eyes that is internal usage due to non-standard name convention.

I'll fix this in the next commit 😄